### PR TITLE
Fix #17796: Export correct system-distance

### LINKED ISSUE #17796: MusicXML exporting: system-distance value is too large for systems that end with a multi-measure rest.
### Issue type

Other type of issue

### Bug description

When exporting MusicXML, the system-distance value in the <print> element of the measure is way too large when the system ends with a multi-measure rest. You can see this in the attached .musicxml file at line 303:
```
    <measure number="15" width="177.03">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>0.00</right-margin>
            </system-margins>
          <system-distance>416.24</system-distance> <----- HERE
          </system-layout>
        </print> 
```

Which is way to big, it should only be around 100, not 400.

### Steps to reproduce

You can easily reproduce the bug by adding a multi-measure rest at the end of the system and then export to uncompressed MusicXML.

### Screenshots/Screen recordings

screenshot

### MuseScore Version

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230651553, revision: github-musescore-musescore-dbe7c6d

### Regression

I don't know

### Operating system

Windows 11

### Additional context

MuseScore project and the MusicXML file:
[files.zip](https://github.com/musescore/MuseScore/files/11605126/files.zip)

### PR DESCRIPTION
**Pull request by rettinghaus**
Fixes #17796

### CODE REVIEW

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -6583,9 +6583,10 @@ void ExportMusicXml::print(const Measure* const m, const int partNr, const int f
 
                 if (mpc.systemStart && !mpc.pageStart) {
                     // see System::layout2() for the factor 2 * score()->spatium()
+                    Measure* prevSystem = mpc.prevMeasure->isMMRest() ? mpc.prevMeasure->mmRestFirst() : mpc.prevMeasure;
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:6586]: Adding a `const` here should fix the build errors

**Comment by cbjeukendrup** [comment]
Looks good to me! It would be ideal if you could also add a test file for this.

**Comment by rettinghaus** [comment]
@cbjeukendrup I'm really willing to add a test, but I don't know how to get it working. Any help would be appreciated.

**Comment by cbjeukendrup** [comment]
@rettinghaus It looks like there are two options:
1. Use `mxmlMscxExportTestRef`. This requires a MuseScore file, and a reference XML file, i.e. `testSystemDistance.mscx` and `testSystemDistance_ref.xml`.
   - However, this method sets `PREF_EXPORT_MUSICXML_EXPORTLAYOUT` to false, while you want it to be true. Easiest solution might be to change the signature of `mxmlMscxExportTestRef` to 
      ```c++
      void Musicxml_Tests::mxmlMscxExportTestRef(const char* file, bool exportLayout = false)
      ```
      and you can doubtlessly fill in the rest from there.
   - When saving a MSCX file in MuseScore 4, it will save a whole folder, but for the purpose of the test you can just take only the mscx file out of that folder.
2. Keep using `mxmlIoTest`, and make sure that the input file in the repo (i.e. `testSystemDistance.xml`) is exactly equal to the expected output file. 
   - you can do this by looking at the diff at https://github.com/musescore/MuseScore/actions/runs/6311023321/job/17134178603?pr=19512#step:9:11130, but note that this diff is the wrong way around; contrary to what you might expect, the `+` lines are what's currently in the file, and the `-` lines are what it should be. (at least, that's my understanding right now...)

Since the purpose here is to test export, and not import/roundtrip, I think option 1 would make slightly more sense, do you agree?

**Comment by rettinghaus** [comment]
@cbjeukendrup Thanks for the tip! Works fine now.

**Review by cbjeukendrup — APPROVED** [review]
Thanks!

**Pull request merged**